src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosFhirDataStore.cs

```diff
@@ -208,7 +205,7 @@ public async Task HardDeleteAsync(ResourceKey key, CancellationToken cancellatio
             }
             catch (DocumentClientException dce)
             {
-                if (dce.Error?.Message?.Contains(GetValue(HttpStatusCode.RequestEntityTooLarge), StringComparison.Ordinal) == true)
+                if (dce.GetSubStatusCode() == HttpStatusCode.RequestEntityTooLarge)
```

Comment:
I think we should throw 429 directly from the stored procedure (althrough it's not defined in ErrorCodes) but I think that's more appropriate. The Cosmos DB doc says RequestEntityTooLarge is used when the document exceeds 2MB but here we are using it when the server rejected the call due to the insufficient RU.